src/ResourceManager/RecoveryServices/Commands.RecoveryServices.Backup/RSBackupVaultCmdletBase.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.RecoveryServices.Backup.Helpers;
+using Microsoft.Azure.Commands.RecoveryServices.Backup.Properties;
+using Microsoft.Azure.Management.RecoveryServices.Backup.Models;
+using AzureRestNS = Microsoft.Rest.Azure;
+using CmdletModel = Microsoft.Azure.Commands.RecoveryServices.Backup.Cmdlets.Models;
+
+namespace Microsoft.Azure.Commands.RecoveryServices.Backup.Cmdlets
+{
+    public abstract class RSBackupVaultCmdletBase : RecoveryServicesBackupCmdletBase
+    {
+        /// <summary>
+        /// The Recovery Services Vault.
+        /// </summary>
+        [Parameter(Mandatory = false, HelpMessage = "The Recovery Services Vault.",
+            ValueFromPipeline = true)]
+        [ValidateNotNullOrEmpty]
+        public ARSVault Vault { get; set; }
+
+        /// <summary>
+        /// Get the job PS model after fetching the job object from the service given the job ID.
+        /// </summary>
+        /// <param name="jobId">ID of the job to be fetched</param>
+        /// <returns></returns>
+        public CmdletModel.JobBase GetJobObject(string jobId, ARSVault vault = null)
```

Comment:
Do the corresponding "GetJobObject/HandleCreatedJob" calls in RecoveryServicesBackupCmdletBase still work with the changes to the adapter? If not, please remove them.

Reply:
Yes, they still work.